s3/src/main/scala/monix/connect/s3/S3.scala

```diff
@@ -517,38 +962,57 @@ object S3 {
+    *
+    * {{{
+    *   import monix.eval.Task
+    *   import monix.reactive.{Observable, Consumer}
     *   import monix.connect.s3.S3
-    *   import monix.execution.Scheduler.Implicits.global
     *   import software.amazon.awssdk.services.s3.S3AsyncClient
     *   import software.amazon.awssdk.services.s3.model.CompleteMultipartUploadResponse
     *   import software.amazon.awssdk.regions.Region.AWS_GLOBAL
     *   import software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider
     *
-    *   //must be propelry configured
-    *   implicit val client = S3AsyncClient.builder.credentialsProvider(DefaultCredentialsProvider.create()).region(AWS_GLOBAL).build()
-    *
-    *   val bucket: String = "sample-bucket"
-    *   val key: String = "sample/key/to/s3/object"
-    *   val content: Array[Byte] = "Hello World!".getBytes
+    *   // must be properly configured
+    *   val s3AsyncClient = S3AsyncClient.builder
+    *     .credentialsProvider(DefaultCredentialsProvider.create())
+    *     .region(AWS_GLOBAL)
+    *     .build()
     *
-    *   val uploadConsumer: Consumer[Array[Byte], CompleteMultipartUploadResponse] = S3.uploadMultipart(bucket, key)
+    *   val s3: S3 = S3.createUnsafe(s3AsyncClient)
     *
-    *   val t: Task[CompleteMultipartUploadResponse] = Observable.pure(content).consumeWith(uploadConsumer)
+    *   val response = Observable.pure(content).consumeWith(s3.uploadMultipart(bucket, key))
     * }}}
     *
-    * @param bucket the bucket name where the object will be stored
-    * @param key the key where the object will be stored.
-    * @param minChunkSize size of the chunks (parts) that will be sent in the http body. (the minimum size is set by default, don't use a lower one)
-    * @param s3AsyncClient implicit instance of the s3 client of type [[S3AsyncClient]]
+    * @param bucket        the bucket name where the object will be stored
+    * @param key           the key where the object will be stored.
+    * @param minChunkSize  size of the chunks (parts) that will be sent in the http body. (the minimum size is set by default, don't use a lower one)
     * @return the confirmation of the multipart whole upload as [[CompleteMultipartUploadResponse]].
     */
   def uploadMultipart(
     bucket: String,
     key: String,
     minChunkSize: Int = awsMinChunkSize,
-    uploadSettings: UploadSettings = DefaultUploadSettings)(
-    implicit
-    s3AsyncClient: S3AsyncClient): Consumer[Array[Byte], CompleteMultipartUploadResponse] =
-    new MultipartUploadSubscriber(bucket, key, minChunkSize, uploadSettings)
+    uploadSettings: UploadSettings = DefaultUploadSettings): Consumer[Array[Byte], CompleteMultipartUploadResponse] = {
+    require(minChunkSize >= domain.awsMinChunkSize, "minChunkSize >= 5242880")
+    new MultipartUploadSubscriber(bucket, key, minChunkSize, uploadSettings, s3Client)
+  }
+
+  /**
+    * Closes the underlying [[S3AsyncClient]].
+    */
+  def close(): Unit = s3Client.close()
```

Comment:
I think we should keep pure API and return `Task[Unit]` here
Alternative, there could be a `closeUnsafe` if it's necessary

Reply:
> > @Avasil would be great to know your thoughts on the new api, you do it :)
> 
> I will take a look in the next days:)

Thanks for reviewing it! :D 
Have addressed the comments